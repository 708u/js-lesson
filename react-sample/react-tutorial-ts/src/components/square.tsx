import React, { FC, MouseEvent } from 'react';
import styled from 'styled-components';

export type mark = 'O' | 'X' | null;

type Props = {
  onClick: (e: MouseEvent) => void,
  value: mark
}

const SquareBtn = styled.button`
  background: #fff;
  border: 1px solid #999;
  float: left;
  font-size: 24px;
  font-weight: bold;
  line-height: 34px;
  height: 34px;
  margin-right: -1px;
  margin-top: -1px;
  padding: 0;
  text-align: center;
  width: 34px;
`;

const Square: FC<Props> = (props) => {
  const { onClick, value } = props;
  return (
    <SquareBtn onClick={ onClick }>
      { value }
    </ SquareBtn>
  );
}

export default Square;
