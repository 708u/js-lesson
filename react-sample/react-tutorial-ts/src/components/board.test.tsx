import renderer from 'react-test-renderer';
import React from 'react';
import Board from 'components/board';
import { mark } from './square';

describe('testing board components', () => {
  it('should be same as a previous snapshot.', () => {
    const squares: mark[] = [
      'O', 'X', 'O', 'X', 'O', 'X', 'O', 'X', 'O',
    ];

    const onClick = (i: number): void => { };
    const component = renderer.create(
      <Board squares={squares} onClick={onClick} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
