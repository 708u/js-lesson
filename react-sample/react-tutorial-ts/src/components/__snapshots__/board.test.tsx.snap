// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing board components should be same as a previous snapshot. 1`] = `
<div>
  <div
    className="sc-gsTCUz gHlRWK"
  >
    <button
      className="sc-bdfBwQ iZlURE"
      onClick={[Function]}
    >
      O
    </button>
    <button
      className="sc-bdfBwQ iZlURE"
      onClick={[Function]}
    >
      X
    </button>
    <button
      className="sc-bdfBwQ iZlURE"
      onClick={[Function]}
    >
      O
    </button>
  </div>
  <div
    className="sc-gsTCUz gHlRWK"
  >
    <button
      className="sc-bdfBwQ iZlURE"
      onClick={[Function]}
    >
      X
    </button>
    <button
      className="sc-bdfBwQ iZlURE"
      onClick={[Function]}
    >
      O
    </button>
    <button
      className="sc-bdfBwQ iZlURE"
      onClick={[Function]}
    >
      X
    </button>
  </div>
  <div
    className="sc-gsTCUz gHlRWK"
  >
    <button
      className="sc-bdfBwQ iZlURE"
      onClick={[Function]}
    >
      O
    </button>
    <button
      className="sc-bdfBwQ iZlURE"
      onClick={[Function]}
    >
      X
    </button>
    <button
      className="sc-bdfBwQ iZlURE"
      onClick={[Function]}
    >
      O
    </button>
  </div>
</div>
`;
